lib/matplotlib/cbook/__init__.py

```diff
@@ -248,14 +248,22 @@ class silent_list(list):
     one will get ::
 
         <a list of 3 Line2D objects>
+
+    If ``self.type`` is None, the type name is obtained from the first item in
+    the list (if any).
     """
+
     def __init__(self, type, seq=None):
         self.type = type
         if seq is not None:
             self.extend(seq)
 
     def __repr__(self):
-        return '<a list of %d %s objects>' % (len(self), self.type)
+        if self.type is not None or self:
```

Comment:
Uh, had to think a bit to understand that ˋor selfˋ is the non-empty check. So much for "explicit is better than implicit".

I think it would be clearer to reverse:
~~~
if not self and self.type is None:
~~~

Reply:
just made it clearer with `len(self) != 0`...